sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -413,17 +413,22 @@ case class DataSource(
     relation
   }
 
-  /** Writes the given [[DataFrame]] out to this [[DataSource]]. */
+  /**
+   * Writes the given [[DataFrame]] out to this [[DataSource]].
+   *
+   * @param isForWriteOnly Whether to just write the data without returning a [[BaseRelation]].
+   */
   def write(
```

Comment:
let's create a new `write` method that returns `Unit`, and rename this `write` to `writeAndRead`, which should be removed eventually.